src/link/Elf.zig

```diff
@@ -1631,7 +1631,20 @@ fn linkWithLLD(self: *Elf, comp: *Compilation) !void {
         if (self.base.options.each_lib_rpath) {
             var test_path = std.ArrayList(u8).init(self.base.allocator);
             defer test_path.deinit();
+
+            const target_info = try std.zig.system.NativeTargetInfo.detect(self.base.allocator, .{});
+            const paths = try std.zig.system.NativePaths.detect(self.base.allocator, target_info);
+
             for (self.base.options.lib_dirs) |lib_dir_path| {
+                var is_target_lib_dir = false;
+                for (paths.lib_dirs.items) |lib_dir| {
+                    if (mem.eql(u8, lib_dir, lib_dir_path)) {
+                        is_target_lib_dir = true;
+                    }
+                }
+                if (is_target_lib_dir) {
+                    continue;
+                }
```

Comment:
There is probably a way to do this better.

Reply:
ended up force pushing twice because I was thrown for a loop around this.

I think this should be the best implementation for this
```zig
                var is_target_lib_dir = false;
                for (paths.lib_dirs.items) |lib_dir| {
                    if (mem.eql(u8, lib_dir, lib_dir_path)) {
                        is_target_lib_dir = true;
                        break;
                    }
                }
                if (is_target_lib_dir) {
                    continue;
                }
```

it might be worth to use labeled loops for this, but I am unsure